picard/util/__init__.py

```diff
@@ -708,3 +709,13 @@ def extract_year_from_date(dt):
             return parse(dt).year
     except (TypeError, ValueError):
         return None
+
+
+def cached_settings(keys, settings=None, config=None):
```

Comment:
Do we need some comments that explain the settings and config parameters i.e. settings to update an existing dict, config to use an existing thread-safe Config object?